aten/src/ATen/core/jit_type.h

```diff
@@ -641,6 +665,7 @@ struct CAFFE2_API DictType : public Type {
 
   static DictTypePtr create(TypePtr key, TypePtr value) {
     switch (key->kind()) {
+      case TypeKind::AnyType:
```

Comment:
Any is not really a valid key type for dictionaries, right? Since it may end up being something not in the list below.

I suppose there's not an easy way to represent empty dicts without another `KeyType` cluttering things, so I think it's fine to leave. How are we ensuring that it's not possible to somehow construct a dictionary with an invalid key type?

Reply:
This check is the at-creation-type check that the dict can't possibly work. There is still a at insertion/lookup-time check that the key is hashable. It is valid to have a Any -> X dictionary. It is just the case that dynamically trying to insert a non-hashable key into it will fail.